cpp/src/groupby/new_groupby.cu

```diff
@@ -190,3 +195,220 @@ gdf_error gdf_group_by(gdf_column* in_key_columns[],
+   */
+  /* ----------------------------------------------------------------------------*/
+gdf_error gdf_group_by_wo_aggregations(int num_data_cols,
+                           	   	   	   gdf_column** data_cols_in,
+									   int num_groupby_cols,
+									   int * groupby_col_indices,
+									   gdf_column** data_cols_out,
+									   gdf_column* group_start_indices,
+                     gdf_context* ctxt)
+{
+  GDF_REQUIRE((nullptr != data_cols_in), GDF_DATASET_EMPTY);
+  GDF_REQUIRE((nullptr != data_cols_in[0]), GDF_DATASET_EMPTY);
+  GDF_REQUIRE((nullptr != data_cols_out), GDF_DATASET_EMPTY);
+  GDF_REQUIRE((nullptr != data_cols_out[0]), GDF_DATASET_EMPTY);
+  GDF_REQUIRE((num_data_cols > 0), GDF_DATASET_EMPTY);
+  GDF_REQUIRE((num_groupby_cols > 0), GDF_DATASET_EMPTY);
+  GDF_REQUIRE((nullptr != groupby_col_indices), GDF_DATASET_EMPTY);
+
+
+  int32_t nrows = data_cols_in[0]->size;
+
+  // setup for order by call
+  bool group_by_keys_contain_nulls = false;
+  std::vector<gdf_column*> orderby_cols_vect(num_groupby_cols);
+  for (int i = 0; i < num_groupby_cols; i++){
+    orderby_cols_vect[i] = data_cols_in[groupby_col_indices[i]];
+    group_by_keys_contain_nulls = group_by_keys_contain_nulls || orderby_cols_vect[i]->null_count > 0;
+  }
+
+  rmm::device_vector<int32_t> sorted_indices(nrows);
+  gdf_column sorted_indices_col;
+  gdf_error status = gdf_column_view(&sorted_indices_col, (void*)(sorted_indices.data().get()), 
+                            nullptr, nrows, GDF_INT32);
+  if (status != GDF_SUCCESS)
+    return status;
+
+  if (ctxt->flag_groupby_include_nulls == 1 || !group_by_keys_contain_nulls){  // SQL style
+  // run order by and get new sort indexes
+    status = gdf_order_by(&orderby_cols_vect[0],             //input columns
+                          nullptr,
+                          num_groupby_cols,                //number of columns in the first parameter (e.g. number of columsn to sort by)
+                          &sorted_indices_col,            //a gdf_column that is pre allocated for storing sorted indices
+                          ctxt);
+    if (status != GDF_SUCCESS)
+      return status;
+
+    // run gather operation to establish new order
+    std::unique_ptr< gdf_table<int32_t> > table_in{new gdf_table<int32_t>{num_data_cols, data_cols_in}};
+    std::unique_ptr< gdf_table<int32_t> > table_out{new gdf_table<int32_t>{num_data_cols, data_cols_out}};
+
+    status = table_in->gather<int32_t>(sorted_indices, *table_out.get());
+    if (status != GDF_SUCCESS)
+      return status;
+
+    status = gdf_group_start_indices(num_data_cols, data_cols_out, num_groupby_cols,
+                      groupby_col_indices, group_start_indices, ctxt);
+
+    return status;
+  } else {  // Pandas style
+
+    int flag_nulls_sort_behavior = ctxt->flag_nulls_sort_behavior;
+    ctxt->flag_nulls_sort_behavior = 2; // overide behaviour to filter out the nulls
```

Comment:
Instead of overwriting the `context` that's passed in, just create a new `gdf_context` and copy the input `gdf_context` into your temporary.

But I guess since it's a C struct you don't get a default copy constructor...